app/routes.py

```diff
@@ -94,8 +95,8 @@ def colorized_icon(
         image = Image.open(fd)
         if image.mode == 'P':
             image = image.convert('RGBA')
-        new_size = int(48 * scale)
-        if new_size != icon_set.get_default_pixel_size():
+        if scale != 1:
+            new_size = int(DEFAULT_ICON_SIZE * scale)
```

Comment:
Here you should take the size from your function not the hardcoded 48px, so the code works if we have in future non 48px icons and also if the icon are not squared:

```python
if scale != 1:
  new_size = icon.get_size().map(s => s * scale)
  image = image.resize(new_size)
```